x-pack/plugins/apm/server/routes/metrics/by_agent/serverless/duration.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { euiLightVars as theme } from '@kbn/ui-theme';
+import { FAAS_BILLED_DURATION } from '../../../../../common/elasticsearch_fieldnames';
+import { LatencyAggregationType } from '../../../../../common/latency_aggregation_types';
+import { getVizColorForIndex } from '../../../../../common/viz_colors';
+import { Setup } from '../../../../lib/helpers/setup_request';
+import { getLatencyTimeseries } from '../../../transactions/get_latency_charts';
+import {
+  fetchAndTransformMetrics,
+  GenericMetricsChart,
+} from '../../fetch_and_transform_metrics';
+import { ChartBase } from '../../types';
+
+const billedDurationAvg = {
+  title: i18n.translate('xpack.apm.agentMetrics.serverless.billedDurationAvg', {
+    defaultMessage: 'Billed Duration',
+  }),
+};
+
+const chartBase: ChartBase = {
+  title: i18n.translate('xpack.apm.agentMetrics.serverless.avgDuration', {
+    defaultMessage: 'Avg. Duration',
+  }),
+  key: 'avg_duration',
+  type: 'linemark',
+  yUnit: 'time',
+  series: {},
+};
+
+async function getTransactionDurationSeries({
+  environment,
+  kuery,
+  setup,
+  serviceName,
+  start,
+  end,
+  searchAggregatedTransactions,
+}: {
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  serviceName: string;
+  start: number;
+  end: number;
+  searchAggregatedTransactions: boolean;
+}): Promise<GenericMetricsChart['series']> {
+  const transactionLatency = await getLatencyTimeseries({
+    environment,
+    kuery,
+    serviceName,
+    transactionType: undefined,
+    transactionName: undefined,
+    setup,
+    searchAggregatedTransactions,
+    latencyAggregationType: LatencyAggregationType.avg,
+    start,
+    end,
+  });
+
+  return [
+    {
+      title: i18n.translate(
+        'xpack.apm.agentMetrics.serverless.transactionDuration',
+        { defaultMessage: 'Transaction Duration' }
+      ),
+      key: 'transaction_duration',
+      type: 'linemark',
+      color: getVizColorForIndex(1, theme),
+      overallValue: transactionLatency.overallAvgDuration ?? 0,
+      data: transactionLatency.latencyTimeseries,
+    },
+  ];
+}
+
+export async function getDuration({
```

Comment:
this too, I'd like to see a more specific name

Reply:
changed it to `serverless_function_latency`